lib/kubeclient/common.rb

```diff
@@ -308,11 +308,20 @@ def get_entity(klass, resource_name, name, namespace = nil, options = {})
       new_entity(result, klass)
     end
 
-    def delete_entity(resource_name, name, namespace = nil)
+    def delete_entity(resource_name, name, namespace = nil, delete_options = {})
+      delete_options_hash = delete_options.to_hash
       ns_prefix = build_namespace_prefix(namespace)
-      handle_exception do
-        rest_client[ns_prefix + resource_name + "/#{name}"]
-          .delete(@headers)
+      payload = delete_options_hash.to_json unless delete_options_hash.empty?
+      _response = handle_exception do
+        rs = rest_client[ns_prefix + resource_name + "/#{name}"]
+        RestClient::Request.execute(
```

Comment:
@ilackarms Please review — does Request.execute require any extra error handling (check status code)?
Or was that only a concern in your PR when used with `block_response`?

Reply:
@cben i don't think it has anything to do with `block_response`; `Request.execute` handles errors differently than `Resource.get`. I assume that's because RestClient makes assumptions about what status codes mean for a `Resource`, but not for a generic `Request` (e.g.: that a 404 should produce an error for a `GET` on a Resource, but a 404 might be a non-error status for a different type of request). `Request.execute` will simply return a response object without raising anything. 

Again I haven't tested it the other way, but my gut feeling is that `Request.execute` is similar to `curl`; as long as the remote server returns an HTTP response of some kind, no error will be raised.

Reply:
@ilackarms  the method will raise an error when the resource is not found.

Reply:
Confirmed experimentally.  Normal `Request.execute` follows redirects and raises on errors.  The high-level `get` and `Resource#get` are thin wrappers around it, no extra handling logic:
https://github.com/rest-client/rest-client/blob/v2.0.2/lib/restclient.rb#L66-L68
https://github.com/rest-client/rest-client/blob/v2.0.2/lib/restclient/resource.rb#L49-L55

It's only with `block_response` that [bypasses normal processing](https://github.com/rest-client/rest-client/blob/v2.0.2/lib/restclient/request.rb#L718-L726).